### PR TITLE
899 ditch old use api

### LINKED ISSUE #899: Refactor reports, anything remaining to use new use api
Enrollment flow is in separate ticket

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #899

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/App/App.tsx [hunk 1]
```diff
@@ -27,20 +27,18 @@ const App: React.FC = () => {
 		orgId: getIdForUser(user, 'org'),
 	};
 
-	const [loading, error, reports] = useApi(
+	const { loading, error, data: reports } = useApi(
 		api => api.apiOrganizationsOrgIdReportsGet(params),
-		[user, cacheInvalidator],
 		{
-			skip: !user,
+			skip: !user || !cacheInvalidator,
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/App/App.tsx:33]: Not 100% sure what this addition to skip is doing

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx [hunk 2]
```diff
@@ -20,23 +20,21 @@ import {
 	serverErrorForField,
 	isBlockingValidationError,
 } from '../../../utils/validations';
-import usePromiseExecution from '../../../hooks/usePromiseExecution';
 import { reportSubmittedAlert, reportSubmitFailAlert } from '../../../utils/stringFormatters';
 import pluralize from 'pluralize';
 import { validationErrorAlert } from '../../../utils/stringFormatters/alertTextMakers';
 import { FormReducer, formReducer, updateData } from '../../../utils/forms/form';
+import useNewUseApi from '../../../hooks/useApi';
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx:27]: Nit: Uses the old name

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ import getFundingSpaceCapacity from '../../../utils/getFundingSpaceCapacity';
 import { validationErrorAlert } from '../../../utils/stringFormatters/alertTextMakers';
 import AlertContext from '../../../contexts/Alert/AlertContext';
 import displayErrorOrWarning from '../../../utils/validations/displayErrorOrWarning';
-import useNewUseApi, { ApiError } from '../../../hooks/newUseApi';
+import useNewUseApi, { ApiError } from '../../../hooks/useApi';
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx:57]: rename

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.test.tsx [hunk 4]
```diff
@@ -20,12 +20,7 @@ describe('enrollment sections', () => {
 			];
 
 			const { findByText } = render(
-				<FamilyIncome.Form
-					siteId={1}
-					enrollment={enrollmentWithValidationErrors}
-					mutate={async () => null}
-					error={null}
-				/>
+				<FamilyIncome.Form siteId={1} enrollment={enrollmentWithValidationErrors} error={null} />
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.test.tsx:23]: this should have come up in review for the enrollment sections (my bad, and feel free to ignore for this PR), but with the new useApi do we want/need to be passing `error` through the enrollment sections?

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.tsx [hunk 5]
```diff
@@ -34,7 +34,7 @@ import { validationErrorAlert } from '../../../utils/stringFormatters/alertTextM
 import AlertContext from '../../../contexts/Alert/AlertContext';
 import { FormReducer, formReducer, updateData } from '../../../utils/forms/form';
 import { DeepNonUndefineable } from '../../../utils/types';
-import useNewUseApi, { ApiError } from '../../../hooks/newUseApi';
+import useNewUseApi, { ApiError } from '../../../hooks/useApi';
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.tsx:37]: rename

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyInfo.tsx [hunk 6]
```diff
@@ -21,7 +21,7 @@ import { validationErrorAlert } from '../../../utils/stringFormatters/alertTextM
 import AlertContext from '../../../contexts/Alert/AlertContext';
 import { FormReducer, formReducer, updateData } from '../../../utils/forms/form';
 import { DeepNonUndefineable } from '../../../utils/types';
-import useNewUseApi, { ApiError } from '../../../hooks/newUseApi';
+import useNewUseApi, { ApiError } from '../../../hooks/useApi';
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyInfo.tsx:24]: rename

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx [hunk 7]
```diff
@@ -109,23 +109,31 @@ export default function ReportSubmitForm({
 		}
 	}, [error, hasAlertedOnError]);
 
-	function _onSubmit() {
-		return mutate(api =>
+	const [attemptingSave, setAttemptingSave] = useState(false);
+	const { loading, error: saveError, data: saveData } = useNewUseApi<CdcReport>(
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx:113]: rename

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx (hunk 2 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitForm.tsx:27]: duplicate import! drop in favor of line 7?

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.test.tsx (hunk 4 above)
Comment (mmazanec22) [reply src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.test.tsx:23]: We still need to for the initial load, or when we change the URL with afterSave and it re-fetches the enrollment

**Pull request merged**